shared-bindings/spitarget/SPITarget.c

```diff
@@ -0,0 +1,182 @@
+//|         ...
+STATIC mp_obj_t spitarget_spi_target_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    spitarget_spi_target_obj_t *self = mp_obj_malloc(spitarget_spi_target_obj_t, &spitarget_spi_target_type);
+    enum { ARG_sck, ARG_mosi, ARG_miso, ARG_ss };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_sck, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_mosi, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_miso, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_ss, MP_ARG_REQUIRED | MP_ARG_OBJ },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    const mcu_pin_obj_t *sck = validate_obj_is_free_pin(args[ARG_sck].u_obj, MP_QSTR_sck);
+    const mcu_pin_obj_t *mosi = validate_obj_is_free_pin(args[ARG_mosi].u_obj, MP_QSTR_mosi);
+    const mcu_pin_obj_t *miso = validate_obj_is_free_pin(args[ARG_miso].u_obj, MP_QSTR_miso);
+    const mcu_pin_obj_t *ss = validate_obj_is_free_pin(args[ARG_ss].u_obj, MP_QSTR_ss);
+
+    common_hal_spitarget_spi_target_construct(self, sck, mosi, miso, ss);
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def deinit(self) -> None:
+//|         """Releases control of the underlying hardware so other classes can use it."""
+//|         ...
+STATIC mp_obj_t spitarget_spi_target_obj_deinit(mp_obj_t self_in) {
+    mp_check_self(mp_obj_is_type(self_in, &spitarget_spi_target_type));
+    spitarget_spi_target_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_spitarget_spi_target_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(spitarget_spi_target_deinit_obj, spitarget_spi_target_obj_deinit);
+
+//|     def __enter__(self) -> SPITarget:
+//|         """No-op used in Context Managers."""
+//|         ...
+//  Provided by context manager helper.
+
+//|     def __exit__(self) -> None:
+//|         """Automatically deinitializes the hardware on context exit. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+STATIC mp_obj_t spitarget_spi_target_obj___exit__(size_t n_args, const mp_obj_t *args) {
+    mp_check_self(mp_obj_is_type(args[0], &spitarget_spi_target_target_type));
+    spitarget_spi_target_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    common_hal_spitarget_spi_target_deinit(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(spitarget_spi_target___exit___obj, 4, 4, spitarget_spi_target_obj___exit__);
+
+//|     def load_packet(self, mosi_packet: bytearray, miso_packet: bytearray) -> None:
+//|         """Queue data for the next SPI transfer from the main device.
+//|         If a packet has already been queued for this SPI bus but has not yet been transferred, an error will be raised.
+//|
+//|         :param bytearray miso_packet: Packet data to be sent from secondary to main on next request."""
```

Comment:
Please document `mosi_packet` too. Is the expectation that this memory isn't mutated after the function call?

Reply:
Documented `mosi_packet`. Yes, that is the expectation and that is how we've run our tests, although I don't think anything would break if the memory was mutated mid-transfer, it would just make it difficult to predict whether the bytes actually transmitted are those placed into the bytearray before the function call or those placed after the function call.